testing/it/http/src/test/java/com/adobe/cq/wcm/core/components/it/http/PageIT.java

```diff
@@ -55,4 +64,47 @@ public void testBrandSlug() throws ClientException {
+        // Theme Color
+        GraniteAssert.assertRegExFind(content, "<meta name=\"theme-color\" content=\"#FF851B\"/>");
+        // Apple Touch Icon
+        GraniteAssert.assertRegExFind(content, "<link rel=\"apple-touch-icon\" href=\".*/content/dam/foo/pwa-logo.png\"/>");
+        // Style sheet for messages
+        GraniteAssert.assertRegExFind(content, "<link rel=\"stylesheet\" href=\".*/etc.clientlibs/core/wcm/components/page/v2/page/clientlibs/site/pwa.*.min.css\" type=\"text/css\">");
+        // Path to service worker
+        GraniteAssert.assertRegExFind(content, "<meta name=\"cq:sw_path\" content=\".*/core-components.simple-pagesw.js\"/>");
+        // Reference to script that registers the service worker
+        GraniteAssert.assertRegExFind(content, "<script src=\".*/etc.clientlibs/core/wcm/components/page/v2/page/clientlibs/site/pwa.*.min.js\"></script>");
+
+        // validate that the sub page also has the PWA properties based on the parent.
+        content = adminAuthor.doGet("/content/core-components/simple-page/simple-subpage.html", 200).getContent();
+        GraniteAssert.assertRegExFind(content, "<link rel=\"manifest\" href=\".*/content/foo/us/en/manifest.webmanifest\" crossorigin=\"use-credentials\"/>");
+        GraniteAssert.assertRegExFind(content, "<meta name=\"theme-color\" content=\"#FF851B\"/>");
+        GraniteAssert.assertRegExFind(content, "<link rel=\"apple-touch-icon\" href=\".*/content/dam/foo/pwa-logo.png\"/>");
+        GraniteAssert.assertRegExFind(content, "<link rel=\"stylesheet\" href=\".*/etc.clientlibs/core/wcm/components/page/v2/page/clientlibs/site/pwa.*.min.css\" type=\"text/css\">");
+        GraniteAssert.assertRegExFind(content, "<meta name=\"cq:sw_path\" content=\".*/core-components.simple-pagesw.js\"/>");
+        GraniteAssert.assertRegExFind(content, "<script src=\".*/etc.clientlibs/core/wcm/components/page/v2/page/clientlibs/site/pwa.*.min.js\"></script>");
+    }
+
+    @Test
+    public void testServiceWorkerConfiguration() throws ClientException {
+        String content = adminAuthor.doGet("/content/core-components/simple-page.sw.js", 200).getContent();
+        Pattern swconfigPattern = Pattern.compile("const swconfig = (?<swconfigjson>\\{.*})", Pattern.MULTILINE);
+        Matcher swconfigMatcher= swconfigPattern.matcher(content);
+        assertTrue(swconfigMatcher.find());
+        Gson gson = new Gson();
+        JsonObject swconfigJson = gson.fromJson(swconfigMatcher.group("swconfigjson"), JsonObject.class);
+        assertNotNull(swconfigJson);
+        assertEquals("staleWhileRevalidate", swconfigJson.get("pwaCachestrategy").getAsString());
+        assertNotNull(swconfigJson.get("pwaPrecache").getAsJsonArray());
+        assertNotNull(swconfigJson.get("pwaCachingpaths").getAsJsonArray());
+        //assertNotNull(swconfigJson.get("pwaOfflineClientlibs").getAsJsonArray());
```

Comment:
Added better handling of JSON as the previous was failing in my local development environment.

Reply:
You might want to expand assertions by walking the expected tree.

Reply:
Done.